### PR TITLE
[Payment][Payum][Pricing] Clean up payment related bundles

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/PayumBundle/spec/Action/Offline/ConvertPaymentActionSpec.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    function it_supports_only_converting_to_array_from_payment(
+        Convert $fromSomethingElseToSomethingElseRequest,
+        Convert $fromPaymentToArrayRequest,
+        PaymentInterface $payment
+    ) {
+        $fromPaymentToArrayRequest->getTo()->willReturn('array');
+        $fromPaymentToArrayRequest->getSource()->willReturn($payment);
+
+        $fromSomethingElseToSomethingElseRequest->getTo()->willReturn(Argument::any());
```
Comment (pamil) [inline src/Sylius/Bundle/PayumBundle/spec/Action/Offline/ConvertPaymentActionSpec.php]: `Argument` class is used for matching arguments, not for being returned as a "something else".

File: src/Sylius/Bundle/PricingBundle/Form/EventListener/BuildPriceableFormSubscriber.php [hunk 2]
```diff
@@ -117,7 +117,7 @@ public function postSubmit(FormEvent $event)
      * @param string        $calculatorType
```
Comment (pamil) [inline src/Sylius/Bundle/PricingBundle/Form/EventListener/BuildPriceableFormSubscriber.php]: These alignments can be deleted too.

File: src/Sylius/Bundle/PricingBundle/Twig/PricingExtension.php [hunk 3]
```diff
@@ -19,7 +19,7 @@
  *
  * @author Paweł Jędrzejewski <pawel@sylius.org>
  */
-class PricingExtension extends \Twig_Extension
+final class PricingExtension extends \Twig_Extension
 {
     /**
      * Templating helper.
```
Comment (pamil) [inline src/Sylius/Bundle/PricingBundle/Twig/PricingExtension.php]: Really helpful comment! :tada:

File: src/Sylius/Bundle/PricingBundle/spec/Form/Extension/PriceableTypeExtensionSpec.php [hunk 4]
```diff
@@ -37,7 +38,7 @@ function it_builds_form(
         $calculatorRegistry,
         $subscriber,
```
Comment (pamil) [inline src/Sylius/Bundle/PricingBundle/spec/Form/Extension/PriceableTypeExtensionSpec.php]: Missing typehints :'(

File: src/Sylius/Bundle/PaymentBundle/Resources/config/services.xml [hunk 5]
```diff
@@ -26,17 +26,17 @@
             <argument>Payment methods resolver</argument>
         </service>
 
-        <service id="sylius.composite_payment_methods_resolver" class="Sylius\Component\Payment\Resolver\CompositeMethodsResolver">
+        <service id="sylius.payment_methods_resolver.composite" class="Sylius\Component\Payment\Resolver\CompositeMethodsResolver">
```
Comment (pamil) [inline src/Sylius/Bundle/PaymentBundle/Resources/config/services.xml]: `sylius.payment_methods_resolver.composite` -> `sylius.payment_methods_resolver` - seems like it is the methods resolver that should be used by third-party code etc.

File: src/Sylius/Bundle/PaymentBundle/Resources/config/services.xml [hunk 6]
```diff
@@ -26,17 +26,17 @@
...
             <argument type="service" id="sylius.registry.payment_methods_resolver" />
         </service>
 
-        <service id="sylius.payment_methods_resolver" class="Sylius\Component\Payment\Resolver\PaymentMethodsResolver">
+        <service id="sylius.payment_methods_resolver.default" class="Sylius\Component\Payment\Resolver\PaymentMethodsResolver">
```
Comment (pamil) [inline src/Sylius/Bundle/PaymentBundle/Resources/config/services.xml]: Add `public="false"`?

File: src/Sylius/Bundle/PaymentBundle/Resources/config/services.xml [hunk 7]
```diff
@@ -26,17 +26,17 @@
...
             <argument type="service" id="sylius.repository.payment_method" />
             <tag name="sylius.payment_method_resolver" type="default" label="Default" />
         </service>
 
         <service id="sylius.form.type.payment_method_choice" class="Sylius\Bundle\PaymentBundle\Form\Type\PaymentMethodChoiceType">
-            <argument type="service" id="sylius.composite_payment_methods_resolver" />
+            <argument type="service" id="sylius.payment_methods_resolver.composite" />
```
Comment (pamil) [inline src/Sylius/Bundle/PaymentBundle/Resources/config/services.xml]: `sylius.payment_methods_resolver.composite` -> `sylius.payment_methods_resolver`

**Comment by pjedrzejewski** [comment]
Thank you Arek! :)

**Pull request merged**